gstd-async/src/lib.rs

```diff
@@ -41,10 +41,28 @@ where
     let waker = waker::empty();
     let mut cx = Context::from_waker(&waker);
 
+    let key_id = gstd::msg::id();
+    let futures = msg::futures(key_id);
+    futures.reset_current();
+
     // Poll
     if let Poll::Ready(v) = future.poll(&mut cx) {
+        // Reset the current message's state
+        futures.clear();
         Some(v)
     } else {
         None
     }
 }
+
+pub fn resume() {
+    let sent_msg_id = gstd::msg::reply_to();
+    let waiting_messages = msg::waiting_messages();
+    if waiting_messages.contains_key(&sent_msg_id) {
```

Comment:
what if not found?

Handle it or leave a TODO + issue

Reply:
We can send a trap I guess. No idea what to do if we have an unexpected reply.

Reply:
Added TODO and an issue: https://github.com/gear-tech/gear/issues/148